site/docs/spark.md

```diff
@@ -258,6 +258,38 @@ ALTER TABLE prod.db.sample DROP COLUMN id
 ALTER TABLE prod.db.sample DROP COLUMN point.z
 ```
 
+### `ALTER TABLE ... ADD PARTITION FIELD`
+
+```sql
+ALTER TABLE prod.db.sample ADD PARTITION FIELD catalog -- identity transform
+ALTER TABLE prod.db.sample ADD PARTITION FIELD bucket(16, id)
+ALTER TABLE prod.db.sample ADD PARTITION FIELD truncate(data, 4)
+ALTER TABLE prod.db.sample ADD PARTITION FIELD years(ts)
+-- use optional AS keyword to specify a custom name for the partition field 
+ALTER TABLE prod.db.sample ADD PARTITION FIELD bucket(16, id) AS shard
+```
+
+### `ALTER TABLE ... DROP PARTITION FIELD`
```

Comment:
Both of these new commands should have a warning that changing partitioning will change the behavior of dynamic writes, which overwrite any partition that is written to. If you partition by days and move to partitioning by hours, overwrites will overwrite hourly partitions but not days anymore. Definitely something to be aware of and a good reason to use `MERGE INTO` or the new `DataFrameWriterV2` API with an explicit overwrite filter.

Reply:
Thanks for the information, I updated a warning block for this. I also added a line at the top Spark feature support table suggesting they need extensions enabled.